Example/SuperAwesomeExampleTests/Common/Components/AdProcessorTest.swift

```diff
@@ -15,8 +15,9 @@ class AdProcessorTests: XCTestCase {
     private let imageFormatUsed = "imgformatused"
     private let mediaFormatUsed = "mediaformatused"
     private let tagFormatUsed = "tagformatused"
+    private let options = ["testKey1": "testValue1", "testKey2": "testValue2"]
 
-    private func testProcess(_ format: CreativeFormatType, _ html: String?, _ vast: String?) {
+    private func testProcess(_ format: CreativeFormatType, _ html: String?, _ vast: String?, _ options: [String: String]?) {
```

Comment:
```suggestion
    private func testProcess(_ format: CreativeFormatType, _ html: String?, _ vast: String?, _ options: [String: Any]?) {
```

Shouldn't the options array contain `Any` so that you can pass an `Int` or a `String`?

Reply:
It's a good point, that would make it more flexible. Was there a reason for the options on the `Configuration` being `[String: String]?` @gunhansancar ?

Reply:
We convert those options to query parameters, so I don't think `Any` is a good idea. We can consider `Codable` maybe? We shouldn't try to cover unrequested demands though. Could you show me what is the use case?

Reply:
The use case is unfortunately being able to pass any additional data via querystring. So we might need to leave it pretty open.

Reply:
Well, if that's the case then we should also update the previous implementation where setting the data on sdk.init as well 👍

Reply:
I noticed that changing to `[String: Any]` meant we lost `Codable` on `AdRequest` so I've added a `PropertyWrapper` to ignore it when it comes to encoding / decoding otherwise it would probably mean another parameter to pass around. Can we foresee any issues with that? I think the only use of `Codable` for that type is in `WebView`? 

I've also currently constrained `Any` to `String` and `Int` as it might be a bit too open otherwise. Do we want to include any other types bearing in mind they will end up as query params, perhaps `Double` or `Bool`?